src/actions/websocket.js

```diff
@@ -552,27 +553,15 @@ function handleChannelDeletedEvent(msg, dispatch, getState) {
     }
 }
 
-function handleChannelUpdatedEvent(msg, dispatch, getState) {
-    let channel;
-    try {
-        channel = msg.data ? JSON.parse(msg.data.channel) : null;
-    } catch (err) {
-        return;
-    }
+async function handleChannelUpdatedEvent(msg, dispatch, getState) {
+    const channelId = msg.data.channel_id;
+    const {data: channel} = await dispatch(getChannel(channelId));
 
-    const entities = getState().entities;
-    const {currentChannelId} = entities.channels;
-    if (channel) {
-        dispatch({
-            type: ChannelTypes.RECEIVED_CHANNEL,
-            data: channel,
-        });
-
-        if (currentChannelId === channel.id) {
-            // Emit an event with the channel received as we need to handle
-            // the changes without listening to the store
-            EventEmitter.emit(WebsocketEvents.CHANNEL_UPDATED, channel);
-        }
+    const {currentChannelId} = getState().entities.channels;
+    if (currentChannelId === channelId) {
+        // Emit an event with the channel received as we need to handle
+        // the changes without listening to the store
+        EventEmitter.emit(WebsocketEvents.CHANNEL_UPDATED, channel);
```

Comment:
Is it no longer necessary to unconditionally emit a `RECEIVED_CHANNEL` event?

Reply:
It's being used by `mattermost-mobile` - https://github.com/mattermost/mattermost-mobile/blob/master/app/components/channel_drawer/channel_drawer.js#L82

Reply:
Did you mean dispatch `RECEIVED_CHANNEL`?  If that's the case, it's dispatched above with [getChannel](https://github.com/mattermost/mattermost-redux/blob/master/src/actions/channels.js#L414)